extra_data/stacking.py

```diff
@@ -208,3 +208,8 @@ def asarray(self):
         for modno, data in self._data.items():
             arr[modno] = data
         return np.moveaxis(arr, 0, self._stack_axis)
+
+    def squeeze(self):
+        """Drop any length-1 dimensions - see numpy.squeeze()"""
+        slices = [0 if d == 1 else slice(None, None) for d in self.shape]
```

Comment:
Thanks for the quick response! Indeed there is a corner case, in which the data contains only one pulse. I tried to write
a mirror implementation of `np.squeeze` and it seems there are something more to consider. The following is my unfinished code

```py
from numpy import AxisError

def squeeze(self, axis=None):
    if axis is None:
         slices = [0 if d == 1 else slice(None, None) for d in self.shape] 
     
     if not isinstance(axis, int):
         raise TypeError("...")

     try:
         if self.shape[axis] != 1:
             raise ValueError(cannot select an axis to squeeze out which has size not equal to one)
     except IndexError as e:
          raise AxisError(str(e))

     ...
     # take care of negative axis     

     # fix the shape and ndim
```

Reply:
Oh, I'd missed that squeeze takes an axis. If I know which dim I want to drop, I tend to do it like `arr[:, 0]` . But it should be easy to add.